fdbserver/LogSystem.h

```diff
@@ -1087,13 +1090,32 @@ struct LogPushData : NonCopyable {
 		next_message_tags.clear();
 	}
 
-	Standalone<StringRef> getMessages(int loc) { return messagesWriter[loc].toValue(); }
+	Standalone<StringRef> getMessages(int loc) {
+		return messagesWriter[loc].toValue();
+	}
+
+	void recordEmptyMessage(int loc, const Standalone<StringRef>& value) {
+		if (!isEmptyMessage[loc]) {
+			BinaryWriter w(AssumeVersion(g_network->protocolVersion()));
+			Standalone<StringRef> v = w.toValue();
+			if (value.size() > v.size()) {
+				isEmptyMessage[loc] = true;
+			}
+		}
+	}
+
+	float getEmptyMessageRatio() const {
```

Comment:
Please add a short comment here explaining what this does and for what this is used.

Reply:
Done.